dali/pipeline/data/tensor_list.h

```diff
@@ -386,6 +427,9 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
   // Tensor that shares the data with this TensorList (valid only
   // if IsDenseTensor returns true)
   Tensor<Backend> * tensor_view_;
+  // This tensor stores view that has different shape than
+  // the original TensorList
+  Tensor<Backend> * tensor_view_reshape_;
```

Comment:
any chance those can be smart ptrs?

Reply:
Made them unique_ptr, but I still want tensor_list to own the memory.